clang/utils/TableGen/ClangProgModelBuiltinEmitter.cpp

```diff
@@ -848,6 +922,228 @@ void BuiltinNameEmitter::EmitQualTypeFinder() {
   OS << "\n} // Bultin2Qual\n";
 }
 
+JSONBuiltinInterfaceEmitter::TypeDesc::TypeDesc(const Record *T) : Rec(T) {
```

Comment:
this should all be constructor member initializers if at all possible.  You'll likely want to make ToOCLAddressSpace a static function in order to do this.